<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "https://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<meta http-equiv="Content-Type" content="text/xhtml;charset=UTF-8"/>
<meta http-equiv="X-UA-Compatible" content="IE=9"/>
<meta name="generator" content="Doxygen 1.8.18"/>
<meta name="viewport" content="width=device-width, initial-scale=1"/>
<title>sfof: /Users/sfarrens/Documents/Codes/github/sfof/sfof/src/exceptions.hpp Source File</title>
<link href="tabs.css" rel="stylesheet" type="text/css"/>
<script type="text/javascript" src="jquery.js"></script>
<script type="text/javascript" src="dynsections.js"></script>
<link href="doxygen.css" rel="stylesheet" type="text/css" />
</head>
<body>
<div id="top"><!-- do not remove this div, it is closed by doxygen! -->
<div id="titlearea">
<table cellspacing="0" cellpadding="0">
 <tbody>
 <tr style="height: 56px;">
  <td id="projectalign" style="padding-left: 0.5em;">
   <div id="projectname">sfof
   </div>
  </td>
 </tr>
 </tbody>
</table>
</div>
<!-- end header part -->
<!-- Generated by Doxygen 1.8.18 -->
<script type="text/javascript" src="menudata.js"></script>
<script type="text/javascript" src="menu.js"></script>
<script type="text/javascript">
/* @license magnet:?xt=urn:btih:cf05388f2679ee054f2beb29a391d25f4e673ac3&amp;dn=gpl-2.0.txt GPL-v2 */
$(function() {
  initMenu('',false,false,'search.php','Search');
});
/* @license-end */</script>
<div id="main-nav"></div>
<div id="nav-path" class="navpath">
  <ul>
<li class="navelem"><a class="el" href="dir_643eb884498f55c55f718c60669cceea.html">sfof</a></li><li class="navelem"><a class="el" href="dir_2563a7ebc8ac4eb214214124d97ad6ec.html">src</a></li>  </ul>
</div>
</div><!-- top -->
<div class="header">
  <div class="headertitle">
<div class="title">exceptions.hpp</div>  </div>
</div><!--header-->
<div class="contents">
<a href="exceptions_8hpp.html">Go to the documentation of this file.</a><div class="fragment"><div class="line"><a name="l00001"></a><span class="lineno">    1</span>&#160;<span class="preprocessor">#ifndef FOF_EXCEPTIONS_HPP</span></div>
<div class="line"><a name="l00002"></a><span class="lineno">    2</span>&#160;<span class="preprocessor">#define FOF_EXCEPTIONS_HPP</span></div>
<div class="line"><a name="l00003"></a><span class="lineno">    3</span>&#160; </div>
<div class="line"><a name="l00004"></a><span class="lineno">    4</span>&#160;<span class="preprocessor">#include &lt;stdexcept&gt;</span></div>
<div class="line"><a name="l00005"></a><span class="lineno">    5</span>&#160;<span class="preprocessor">#include &lt;string&gt;</span></div>
<div class="line"><a name="l00006"></a><span class="lineno">    6</span>&#160; </div>
<div class="line"><a name="l00008"></a><span class="lineno"><a class="line" href="class_bad_argument_exception.html">    8</a></span>&#160;<span class="keyword">class </span><a class="code" href="class_bad_argument_exception.html">BadArgumentException</a> : <span class="keyword">public</span> std::invalid_argument</div>
<div class="line"><a name="l00009"></a><span class="lineno">    9</span>&#160;{</div>
<div class="line"><a name="l00010"></a><span class="lineno">   10</span>&#160;<span class="keyword">public</span>:</div>
<div class="line"><a name="l00011"></a><span class="lineno">   11</span>&#160;    <a class="code" href="class_bad_argument_exception.html#abbb1df5cc1232157175cd7132d9a874b">BadArgumentException</a>(<span class="keyword">const</span> std::string&amp; arg_name, <span class="keyword">const</span> std::string&amp; expected);</div>
<div class="line"><a name="l00012"></a><span class="lineno">   12</span>&#160;    <a class="code" href="class_bad_argument_exception.html#abbb1df5cc1232157175cd7132d9a874b">BadArgumentException</a>(<span class="keyword">const</span> std::string&amp; context,</div>
<div class="line"><a name="l00013"></a><span class="lineno">   13</span>&#160;            <span class="keyword">const</span> std::string&amp; arg_name, <span class="keyword">const</span> std::string&amp; expected);</div>
<div class="line"><a name="l00014"></a><span class="lineno"><a class="line" href="class_bad_argument_exception.html#ac2672f775f07aca98b7507844d183602">   14</a></span>&#160;    <span class="keyword">virtual</span> <a class="code" href="class_bad_argument_exception.html#ac2672f775f07aca98b7507844d183602">~BadArgumentException</a>() throw(){}</div>
<div class="line"><a name="l00015"></a><span class="lineno">   15</span>&#160;    <span class="keyword">virtual</span> <span class="keyword">const</span> <span class="keywordtype">char</span>* <a class="code" href="class_bad_argument_exception.html#acf2560d14348b3b253535392ffb3bc15">what</a>() <span class="keyword">const</span> throw();</div>
<div class="line"><a name="l00016"></a><span class="lineno">   16</span>&#160;protected:</div>
<div class="line"><a name="l00017"></a><span class="lineno"><a class="line" href="class_bad_argument_exception.html#a8c926ddc32d625cb7a311d688bc6c653">   17</a></span>&#160;    std::<span class="keywordtype">string</span> <a class="code" href="class_bad_argument_exception.html#a8c926ddc32d625cb7a311d688bc6c653">message</a>;</div>
<div class="line"><a name="l00018"></a><span class="lineno">   18</span>&#160; </div>
<div class="line"><a name="l00019"></a><span class="lineno">   19</span>&#160;};</div>
<div class="line"><a name="l00020"></a><span class="lineno">   20</span>&#160; </div>
<div class="line"><a name="l00022"></a><span class="lineno"><a class="line" href="class_domain_exception.html">   22</a></span>&#160;class <a class="code" href="class_domain_exception.html">DomainException</a> : public std::domain_error</div>
<div class="line"><a name="l00023"></a><span class="lineno">   23</span>&#160;{</div>
<div class="line"><a name="l00024"></a><span class="lineno">   24</span>&#160;<span class="keyword">public</span>:</div>
<div class="line"><a name="l00025"></a><span class="lineno">   25</span>&#160;    <a class="code" href="class_domain_exception.html">DomainException</a>(<span class="keyword">const</span> std::string&amp; context, <span class="keyword">const</span> std::string&amp; report);</div>
<div class="line"><a name="l00026"></a><span class="lineno"><a class="line" href="class_domain_exception.html#a03d4bf65405acac31c54ca621a8a7480">   26</a></span>&#160;    <span class="keyword">virtual</span> <a class="code" href="class_domain_exception.html#a03d4bf65405acac31c54ca621a8a7480">~DomainException</a>() throw(){}</div>
<div class="line"><a name="l00027"></a><span class="lineno">   27</span>&#160;    <span class="keyword">virtual</span> <span class="keyword">const</span> <span class="keywordtype">char</span>* <a class="code" href="class_bad_argument_exception.html#acf2560d14348b3b253535392ffb3bc15">what</a>() <span class="keyword">const</span> throw();</div>
<div class="line"><a name="l00028"></a><span class="lineno">   28</span>&#160;protected:</div>
<div class="line"><a name="l00029"></a><span class="lineno"><a class="line" href="class_domain_exception.html#a1540ab215efc1b8e1096c74317c3b1a3">   29</a></span>&#160;    std::<span class="keywordtype">string</span> <a class="code" href="class_bad_argument_exception.html#a8c926ddc32d625cb7a311d688bc6c653">message</a>;</div>
<div class="line"><a name="l00030"></a><span class="lineno">   30</span>&#160; </div>
<div class="line"><a name="l00031"></a><span class="lineno">   31</span>&#160;};</div>
<div class="line"><a name="l00032"></a><span class="lineno">   32</span>&#160; </div>
<div class="line"><a name="l00034"></a><span class="lineno"><a class="line" href="class_runtime_exception.html">   34</a></span>&#160;class <a class="code" href="class_runtime_exception.html">RuntimeException</a> : public std::runtime_error</div>
<div class="line"><a name="l00035"></a><span class="lineno">   35</span>&#160;{</div>
<div class="line"><a name="l00036"></a><span class="lineno">   36</span>&#160;<span class="keyword">public</span>:</div>
<div class="line"><a name="l00037"></a><span class="lineno">   37</span>&#160;    <a class="code" href="class_runtime_exception.html">RuntimeException</a>(<span class="keyword">const</span> std::string&amp; context, <span class="keyword">const</span> std::string&amp; member_name,</div>
<div class="line"><a name="l00038"></a><span class="lineno">   38</span>&#160;              <span class="keyword">const</span> std::string&amp; expected);</div>
<div class="line"><a name="l00039"></a><span class="lineno">   39</span>&#160;    <a class="code" href="class_runtime_exception.html">RuntimeException</a>(<span class="keyword">const</span> std::string&amp; context, <span class="keyword">const</span> std::string&amp; error);</div>
<div class="line"><a name="l00040"></a><span class="lineno"><a class="line" href="class_runtime_exception.html#a2ace1c1edb92f8c7d870452dfb059a53">   40</a></span>&#160;    <span class="keyword">virtual</span> <a class="code" href="class_runtime_exception.html#a2ace1c1edb92f8c7d870452dfb059a53">~RuntimeException</a>() throw(){}</div>
<div class="line"><a name="l00041"></a><span class="lineno">   41</span>&#160;    <span class="keyword">virtual</span> <span class="keyword">const</span> <span class="keywordtype">char</span>* <a class="code" href="class_bad_argument_exception.html#acf2560d14348b3b253535392ffb3bc15">what</a>() <span class="keyword">const</span> throw();</div>
<div class="line"><a name="l00042"></a><span class="lineno">   42</span>&#160;protected:</div>
<div class="line"><a name="l00043"></a><span class="lineno"><a class="line" href="class_runtime_exception.html#afac221a13d2e8db26675156cf673bead">   43</a></span>&#160;    std::<span class="keywordtype">string</span> <a class="code" href="class_bad_argument_exception.html#a8c926ddc32d625cb7a311d688bc6c653">message</a>;</div>
<div class="line"><a name="l00044"></a><span class="lineno">   44</span>&#160; </div>
<div class="line"><a name="l00045"></a><span class="lineno">   45</span>&#160;};</div>
<div class="line"><a name="l00046"></a><span class="lineno">   46</span>&#160; </div>
<div class="line"><a name="l00047"></a><span class="lineno">   47</span>&#160;<span class="preprocessor">#endif</span></div>
</div><!-- fragment --></div><!-- contents -->
<div class="ttc" id="aclass_bad_argument_exception_html_ac2672f775f07aca98b7507844d183602"><div class="ttname"><a href="class_bad_argument_exception.html#ac2672f775f07aca98b7507844d183602">BadArgumentException::~BadArgumentException</a></div><div class="ttdeci">virtual ~BadArgumentException()</div><div class="ttdef"><b>Definition:</b> exceptions.hpp:14</div></div>
<div class="ttc" id="aclass_runtime_exception_html_a2ace1c1edb92f8c7d870452dfb059a53"><div class="ttname"><a href="class_runtime_exception.html#a2ace1c1edb92f8c7d870452dfb059a53">RuntimeException::~RuntimeException</a></div><div class="ttdeci">virtual ~RuntimeException()</div><div class="ttdef"><b>Definition:</b> exceptions.hpp:40</div></div>
<div class="ttc" id="aclass_runtime_exception_html"><div class="ttname"><a href="class_runtime_exception.html">RuntimeException</a></div><div class="ttdoc">Class for runtime exceptions.</div><div class="ttdef"><b>Definition:</b> exceptions.hpp:35</div></div>
<div class="ttc" id="aclass_domain_exception_html_a03d4bf65405acac31c54ca621a8a7480"><div class="ttname"><a href="class_domain_exception.html#a03d4bf65405acac31c54ca621a8a7480">DomainException::~DomainException</a></div><div class="ttdeci">virtual ~DomainException()</div><div class="ttdef"><b>Definition:</b> exceptions.hpp:26</div></div>
<div class="ttc" id="aclass_bad_argument_exception_html_abbb1df5cc1232157175cd7132d9a874b"><div class="ttname"><a href="class_bad_argument_exception.html#abbb1df5cc1232157175cd7132d9a874b">BadArgumentException::BadArgumentException</a></div><div class="ttdeci">BadArgumentException(const std::string &amp;arg_name, const std::string &amp;expected)</div><div class="ttdef"><b>Definition:</b> exceptions.cpp:3</div></div>
<div class="ttc" id="aclass_domain_exception_html"><div class="ttname"><a href="class_domain_exception.html">DomainException</a></div><div class="ttdoc">Class for domain exceptions.</div><div class="ttdef"><b>Definition:</b> exceptions.hpp:23</div></div>
<div class="ttc" id="aclass_bad_argument_exception_html_a8c926ddc32d625cb7a311d688bc6c653"><div class="ttname"><a href="class_bad_argument_exception.html#a8c926ddc32d625cb7a311d688bc6c653">BadArgumentException::message</a></div><div class="ttdeci">std::string message</div><div class="ttdef"><b>Definition:</b> exceptions.hpp:17</div></div>
<div class="ttc" id="aclass_bad_argument_exception_html_acf2560d14348b3b253535392ffb3bc15"><div class="ttname"><a href="class_bad_argument_exception.html#acf2560d14348b3b253535392ffb3bc15">BadArgumentException::what</a></div><div class="ttdeci">virtual const char * what() const</div><div class="ttdef"><b>Definition:</b> exceptions.cpp:17</div></div>
<div class="ttc" id="aclass_bad_argument_exception_html"><div class="ttname"><a href="class_bad_argument_exception.html">BadArgumentException</a></div><div class="ttdoc">Class for bad argument exceptions.</div><div class="ttdef"><b>Definition:</b> exceptions.hpp:9</div></div>
<!-- start footer part -->
<hr class="footer"/><address class="footer"><small>
Generated on Sat May 30 2020 10:16:29 for sfof by &#160;<a href="http://www.doxygen.org/index.html">
<img class="footer" src="doxygen.png" alt="doxygen"/>
</a> 1.8.18
</small></address>
</body>
</html>
